openedx_external_enrollments/tests/tests_tasks.py

```diff
@@ -48,3 +48,47 @@ def test_refresh_viper_api_keys_with_400_status_code(
 
         refresh_api_keys_mock.assert_called_once()
         refresh_viper_api_keys_task_retry_mock.assert_called_once()
+
+
+class TestPathstreamTask(unittest.TestCase):
+    """Test class for Pathstream tasks."""
+
+    @patch('openedx_external_enrollments.tasks.run_pathstream_task.retry')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.pathstream_external_enrollment'
+        '.PathstreamExternalEnrollment.execute_upload'
+    )
+    def test_run_pathstream_task_completed(self, execute_upload_mock, run_pathstream_task_retry_mock):
+        """
+        This test checks the execution of run_pathstream_task when PathstreamExternalEnrollment.execute_upload
+        returns a successful response.
+        """
+        execute_upload_mock.return_value = True, 'execute_upload completed'
+        expected_result = {'message': 'execute_upload completed'}
+
+        result = run_pathstream_task()  # pylint: disable=no-value-for-parameter
+
+        execute_upload_mock.assert_called_once()
+        run_pathstream_task_retry_mock.assert_not_called()
+        self.assertEqual(result, expected_result)
+
+    @patch('openedx_external_enrollments.tasks.run_pathstream_task.retry')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.pathstream_external_enrollment'
+        '.PathstreamExternalEnrollment.execute_upload'
+    )
+    def test_run_pathstream_task_not_completed(self, execute_upload_mock, run_pathstream_task_retry_mock):
+        """
+        This test checks the execution of run_pathstream_task when PathstreamExternalEnrollment.execute_upload
+        returns a failed response.
+        """
+        execute_upload_mock.return_value = False, 'test-ExecutionError-message'
+        expected_result = {'message': 'test-ExecutionError-message'}
+
+        with self.assertRaises(Exception):
```

Comment:
You are right that's the error I should catch. I will not replace just the error but the scope of this test, because what I wanted to test is that our custom exception is raised so the task could retry. @Squirrel18